frontend/src/metabase/models/containers/ModelDetailPage/ModelDetailPage.unit.spec.tsx

```diff
@@ -0,0 +1,352 @@
+import React from "react";
+import nock from "nock";
+import userEvent from "@testing-library/user-event";
+
+import {
+  fireEvent,
+  renderWithProviders,
+  getIcon,
+  queryIcon,
+  screen,
+  waitFor,
+  waitForElementToBeRemoved,
+  within,
+} from "__support__/ui";
+import { PRODUCTS, PEOPLE } from "__support__/sample_database_fixture";
+import {
+  setupCardEndpoints,
+  setupCollectionEndpoints,
+  setupDatabasesEndpoints,
+  setupTableEndpoints,
+} from "__support__/server-mocks";
+
+import Models from "metabase/entities/questions";
+
+import type { Card, Collection, Field } from "metabase-types/api";
+import { createMockCollection, createMockUser } from "metabase-types/api/mocks";
+
+import type Question from "metabase-lib/Question";
+import {
+  getStructuredModel,
+  getNativeModel,
+  getSavedStructuredQuestion,
+  getSavedNativeQuestion,
+} from "metabase-lib/mocks";
+
+import ModelDetailPage from "./ModelDetailPage";
+
+console.warn = jest.fn();
+
+// eslint-disable-next-line react/display-name
+jest.mock("metabase/core/components/Link", () => ({ to, ...props }: any) => (
```

Comment:
WDYT about using `withRouter` instead? https://github.com/metabase/metabase/blob/6d3f59259485913693b8a246c49260fed7767330/frontend/test/__support__/ui.tsx#L41

Reply:
I'm not sure how to render it with a router here. We're testing the top-level component that takes a model ID from `props.params.slug` and `react-router` seem to overwrite whatever I pass a prop here